test/integration/builder/additional.js

```diff
@@ -961,32 +919,27 @@ module.exports = function(knex) {
       const queryTimeout = 10;
       const secondQueryTimeout = 11;
 
-      return getTestQuery()
-        .timeout(queryTimeout, { cancel: true })
-        .then(function() {
-          throw new Error("Shouldn't have gotten here.");
-        })
-        .catch(function(error) {
-          expect(_.pick(error, 'timeout', 'name', 'message')).to.deep.equal({
-            timeout: queryTimeout,
-            name: 'error',
-            message: `After query timeout of ${queryTimeout}ms exceeded, cancelling of query failed.`,
-          });
-        })
-        .then(() => {
-          knexPrototype._wrappedCancelQueryCall = originalWrappedCancelQueryCall;
-          return getTestQuery().timeout(secondQueryTimeout, { cancel: true });
-        })
-        .catch(function(error) {
-          expect(_.pick(error, 'timeout', 'name', 'message')).to.deep.equal({
-            timeout: secondQueryTimeout,
-            name: 'KnexTimeoutError',
-            message: `Defined query timeout of ${secondQueryTimeout}ms exceeded when running query.`,
-          });
-        })
-        .finally(() => {
-          return knexDb.destroy();
+      try {
+        await expect(
+          getTestQuery().timeout(queryTimeout, { cancel: true })
+        ).to.be.eventually.rejected.and.deep.include({
+          timeout: queryTimeout,
+          name: 'error',
+          message: `After query timeout of ${queryTimeout}ms exceeded, cancelling of query failed.`,
         });
+
+        knexPrototype._wrappedCancelQueryCall = originalWrappedCancelQueryCall;
+
+        await expect(
+          getTestQuery().timeout(secondQueryTimeout, { cancel: true })
+        ).to.be.eventually.rejected.and.deep.include({
+          timeout: secondQueryTimeout,
+          name: 'KnexTimeoutError',
+          message: `Defined query timeout of ${secondQueryTimeout}ms exceeded when running query.`,
+        });
+      } finally {
+        await knexDb.destroy();
```

Comment:
It looks like there are several lines between:

1. Instantiating the `knexDb` instance, and
2. The beginning of the `try{} finally{}` code

Which means: if a failure occurs anywhere between these steps, then `knexDb` will never be destroyed.  So, we should probably begin the `try{}` statement a bit earlier.  Ex:

```
const knexDb = new Knex(knexConfig);

// Notice: the `try{}` begins immediately after instantiating `knexDb`
try {
  // Notice: this line used to be outside of the `try{}`
  const rawTestQueries = {
      pg: (sleepSeconds) => `SELECT pg_sleep(${sleepSeconds})`,
  };

  // ... more lines ...
} finally {
  await knexDb.destroy();
}
```

Reply:
As i can see, lines between try block and knex instantiation dont throws, except if
I move new Knex after if

Reply:
Given that this is a test case, it's not really a huge concern.  But, generally speaking, it's best to establish "object disposal guarantees" immediately after instantiating an object.  For example, consider the following scenario:

```
const connection = await createConnection();

// But... we haven't established the connection disposal guarantee yet!
doSomethingwith(connection);

// At last, we're setting up the connection disposal guarantee
try {
  // Do stuff...
} finally {
  await connection.dispose();
}
```

Even if `doSomethingwith(connection)` is safe today, it might start throwing exceptions in a future version of the code.  This will then introduce a connection leak into the code, which might prove to be very difficult to find.